PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/sparkserverless/serverexplore/ui/SparkServerlessProvisionDialog.java

```diff
@@ -0,0 +1,206 @@
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.microsoft.azure.sparkserverless.serverexplore.ui;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.DialogWrapper;
+import com.microsoft.azure.hdinsight.common.mvc.SettableControl;
+import com.microsoft.azure.hdinsight.sdk.common.azure.serverless.AzureSparkServerlessAccount;
+import com.microsoft.azure.hdinsight.sdk.common.azure.serverless.AzureSparkServerlessClusterManager;
+import com.microsoft.azure.sparkserverless.serverexplore.SparkServerlessClusterProvisionCtrlProvider;
+import com.microsoft.azure.sparkserverless.serverexplore.SparkServerlessClusterProvisionSettingsModel;
+import com.microsoft.azure.sparkserverless.serverexplore.sparkserverlessnode.SparkServerlessADLAccountNode;
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import com.microsoft.intellij.rxjava.IdeaSchedulers;
+import org.apache.commons.lang3.math.NumberUtils;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.awt.event.FocusAdapter;
+import java.awt.event.FocusEvent;
+import java.util.Arrays;
+
+public class SparkServerlessProvisionDialog extends DialogWrapper
+        implements SettableControl<SparkServerlessClusterProvisionSettingsModel> {
+    @NotNull
+    private SparkServerlessClusterProvisionCtrlProvider ctrlProvider;
+
+    @NotNull
+    private SparkServerlessADLAccountNode adlAccountNode;
+
+    @NotNull
+    private AzureSparkServerlessAccount account;
+
+    private JTextField clusterNameField;
+    private JTextField adlAccountField;
+    private JTextField userStorageAccountField;
+    private JTextField previousSparkEventsField;
+    private JTextField masterCoresField;
+    private JTextField masterMemoryField;
+    private JTextField workerCoresField;
+    private JTextField workerMemoryField;
+    private JTextField workerNumberOfContainersField;
+    private JTextField availableAUField;
+    private JTextField totalAUField;
+    private JTextField calculatedAUField;
+    private JLabel masterMemoryLabel;
+    private JLabel masterCoresLabel;
+    private JLabel clusterNameLabel;
+    private JLabel adlAccountLabel;
+    private JLabel previousSparkEventsLabel;
+    private JLabel availableAULabel;
+    private JLabel calculatedAULabel;
+    private JLabel workerCoresLabel;
+    private JLabel workerMemoryLabel;
+    private JLabel workerNumberOfContainersLabel;
+    private JTextField errorMessageField;
+    private JPanel provisionDialogPanel;
+    private JButton refreshButton;
+    private JLabel userStorageAccountLabel;
+
+    public SparkServerlessProvisionDialog(@NotNull SparkServerlessADLAccountNode adlAccountNode,
+                                          @NotNull AzureSparkServerlessAccount account) {
+        // TODO: canBeParent
```

Comment:
What the TODO means?

Reply:
`canBeParent` is the second parameter of parent class's constructor. I will remove this note.

---
In reply to: [191115309](https://github.com/Microsoft/azure-tools-for-java/pull/1665#discussion_r191115309) [](ancestors = 191115309)